src/main/java/uk/nhs/tis/trainee/notifications/api/TraineeHistoryResource.java

```diff
@@ -91,30 +88,111 @@ ResponseEntity<List<HistoryDto>> getTraineeHistory(
    * Get the historic notification with the given ID.
    *
    * @param notificationId The ID of the notification to get.
+   * @param token          The authorization token from the request header.
    * @return The found notification.
    */
   @GetMapping("/message/{notificationId}")
   ResponseEntity<String> getHistoricalMessage(@PathVariable String notificationId,
       @RequestHeader(HttpHeaders.AUTHORIZATION) String token) {
     log.info("Retrieving notification message for the authorized trainee.");
+    String traineeId = getTraineeId(token);
+
+    if (traineeId == null) {
+      return ResponseEntity.badRequest().build();
+    }
+
+    log.info("Rebuilding message for notification {}.", notificationId);
+    Optional<String> message = service.rebuildMessage(traineeId, notificationId);
+
+    return message.map(msg -> ResponseEntity.ok().contentType(MediaType.TEXT_HTML).body(msg))
+        .orElseGet(() -> ResponseEntity.notFound().build());
+  }
+
+  /**
+   * Update the status of a notification to set it to "ARCHIVED".
+   *
+   * @param notificationId The ID of the notification to mark as archived.
+   * @param token          The authorization token from the request header.
+   * @return The updated notification history.
+   */
+  @PutMapping("/notification/{notificationId}/archive")
```

Comment:
I suppose its quite reasonable not to have an un-archive endpoint, but just confirming that this was considered.

Reply:
Didn't think that far ahead, not really sure how Archive will work yet